src/server/rdb_load.h

```diff
@@ -172,6 +172,8 @@ class RdbLoader : protected RdbLoaderBase {
     return load_time_;
   }
 
+  std::function<void()> fullsyncb;
+
```

Comment:
How should the loader be extended to react to events? Callback?

Reply:
which events? what are you trying to do? (I have not reviewed the code yet).

Reply:
to fullsync cut opcodes. The loader only finishes and unblocks once it has received the EOF code, but we need to record the FULLSYNC_CUT code earlier